src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs

```diff
@@ -91,10 +92,24 @@ protected override void ExecuteCore()
 
             HashSet<string> unrecognizedRuntimeIdentifiers = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
+            bool windowsOnlyErrorLogged = false;
             foreach (var knownFrameworkReference in knownFrameworkReferencesForTargetFramework)
             {
                 frameworkReferenceMap.TryGetValue(knownFrameworkReference.Name, out ITaskItem frameworkReference);
 
+                // Handle Windows-only frameworks on non-Windows platforms
+                if (knownFrameworkReference.IsWindowsOnly &&
+                    !RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    // It is an error to reference the framework from non-Windows
+                    if (!windowsOnlyErrorLogged && frameworkReference != null)
+                    {
+                        Log.LogError(Strings.WindowsDesktopFrameworkRequiresWindows);
+                        windowsOnlyErrorLogged = true;
+                    }
+                    continue;
```

Comment:
Add a comment to the continue statement explaining that there wasn't a direct reference to this framework and we skip the rest of the loop so that we don't pre-emptively download it in case of transitive references.

Reply:
Will add.